src/main/java/com/aws/greengrass/componentmanager/plugins/DockerImageDownloader.java

```diff
@@ -188,4 +137,107 @@ public boolean canUnarchiveArtifact() {
         // Not applicable for docker images since docker engine abstracts this
         return false;
     }
+
+    private void checkDownloadPrerequisites() throws PackageDownloadException {
+        // Check that Docker engine is installed
+        if (!dockerClient.dockerInstalled()) {
+            throw new PackageDownloadException("Docker engine is not installed on the device, please ensure it's "
+                    + "installed and redo the deployment");
+        }
+    }
+
+    private File performDownloadSteps(Image image) throws PackageDownloadException, InterruptedException {
+        AtomicBoolean credentialRefreshNeeded = new AtomicBoolean(false);
+        do {
+            // Attempt getting credentials and login only for public registries, currently we only
+            // support private registries in ECR so assume others are public.
```

Comment:
Yes, this will be different, you'd need to get the static credential from a different relevant place, i.e. SecretsManager, and that will introduce a new condition here, but will not cause any significant refactoring, I changed this a little bit to make this more clear